backend/apps/vacancies/models.py

```diff
@@ -0,0 +1,131 @@
+from apps.accounts.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from taggit.managers import TaggableManager
+
+
+class EmploymentChoices(models.TextChoices):
+
+    FULL_TIME = "FT", _("Full time")
+    PART_TIME = "PT", _("Part time")
+    OTHER = "OT", _("Other")
+
+
+class EnglishLevelChoices(models.TextChoices):
+
+    UNKNOWN = "A0", _("Unknown")
+    BEGINNER = "A1", _("Beginner")
+    ELEMENTARY = "A2", _("Elementary")
+    INTERMEDIATE = "B1", _("Intermediate")
+    UPPER_INTERMEDIATE = "B2", _("Upper-intermediate")
+    ADVANCED = "C1", _("Advanced")
+    PROFICIENCY = "C2", _("Proficiency")
+
+
+class Currency(models.Model):
+
+    currency_title = models.CharField(_("Currency name"), max_length=64)
+    currency_code = models.CharField(_("Currency code"), max_length=3)
+
+    def __str__(self):
+        return f"{self.currency_code.upper()}: {self.currency_title}"
+
+
+class Vacancy(models.Model):
+
+    """
+    This class represents a Vacancy model, which can be used to create
+    vacancirs, and manage them.
+
+    Attributes
+    ----------
+    title : str
+        a title of vacancy
+    keywords : str
+        a tags with technologies
+    type_of_employment : str
+        a type of employment
+    location : str
+        city and country of work, remote work
+    english_level : str
+        a english_level
+    min_experience : float
+        a minimum experince by vacancy
+    start_date : date
+        recruitment start date
+    end_date : date
+        recruitment end date
+    description : str
+        about vacancy
+    priority : str
+        priority in vacancies
+    salary_max : int
+        maximum salary
+    min_max : int
+        minimum salary
+    author : str
+        author by vacancy
+    contact_person : str
+        contact person y vacancy
+    is_active : boolean
+        if vacancy is active
+    is_salary_show : boolean
+        if we need in vacancy show salary
+    """
+
+    title = models.CharField(_("Title"), max_length=64, db_index=True)
+    keywords = TaggableManager()
+    type_of_employment = models.CharField(
+        _("Type of employment"), choices=EmploymentChoices.choices, db_index=True, max_length=32
+    )
+    location = models.CharField(_("Location"), max_length=128, db_index=True)
+    english_level = models.CharField(
+        _("English level"), choices=EnglishLevelChoices.choices, db_index=True, max_length=32
+    )
+    min_experience = models.FloatField(_("Minimum experience in years"), db_index=True, blank=True)
+    end_date = models.DateField(_("Recruitment end date"))
+    start_date = models.DateField(_("Recruitment start date"))
+    description = models.TextField(_("Description"), max_length=2048)
+    priority = models.TextField(_("Priority"), max_length=512)
```

Comment:
Why is it a text field? Maybe put it in choices as well? If I correctly understand the meaning of the field, though...

Reply:
I also think the choices would be better in here.
Something like: "urgent, high, medium, low".

Reply:
@Voorhees2019 @pazuzu-ua 
As far as I understand, the priority is what is most important in the candidates, what will be an advantage among other candidates, etc.

Reply:
@LV-coding 
Maybe I didn't get a thing...
I thought that was a priority of hiring after a review or something similar.

Reply:
Maybe link it to some field in Candidate model then? Create some model 'Qualities' or something.